src/UI/Components/Diagram/helpers.ts

```diff
@@ -220,3 +221,109 @@ const doesElementsIsEmbeddedWithExhaustedConnections = (
   }
   return false;
 };
+
+/**
+ * Function that will merge state from Instance Composer to proper object for order_api endpoint.
+ * Instance composer state is being split into multiple objects that could be embedded into other available, so we need to recursively
+ * go through all of them to group, and sort them
+ * @param instances all of the instances that were created/edited in the instance, not including the one passed in second parameter
+ * @param instance instance which is used taken into consideration as the parent of the possible embedded
+ * @param isEmbedded boolean informing whether instance passed is embedded or not
+ * @returns
+ */
+export const shapesDataTransform = (
+  instances: InstanceForApi[],
+  instance: InstanceForApi,
+  isEmbedded = false,
+) => {
+  let areEmbeddedEdited = false;
+  const matchingInstances = instances.filter(
+    (checkedInstance) => checkedInstance.embeddedTo === instance.instance_id,
+  );
+
+  const notMatchingInstances = instances.filter(
+    (checkedInstance) => checkedInstance.embeddedTo !== instance.instance_id,
+  );
+
+  //iterate through matching (embedded)instances and group them according to property type to be able to put them in the Array if needed at once
+  const groupedEmbedded: { [key: string]: InstanceForApi[] } =
```

Comment:
I know key is a generic often used, but in our case, I think it would be nicer to use what will actually be used as key. If I'm not mistaken, this would be the instanceId or serviceId, it will make more sense when we come back later on that.